### PR TITLE
Stylish haskell updates

### PR DESCRIPTION
**Pull request by newhoggy**
* Disabled record formatting because it is compatible with our coding style and worse, it deletes some comments.
* Disabled RHS alignment rules.
* Multiline import lists

Please see https://github.com/input-output-hk/cardano-node/pull/1576 to see the kinds of changes to expect

### CODE REVIEW

File: cardano-api/src/Cardano/Api/Protocol.hs [hunk 1]
```diff
@@ -1,13 +1,13 @@
-{-# LANGUAGE ConstraintKinds     #-}
-{-# LANGUAGE DeriveAnyClass      #-}
-{-# LANGUAGE DeriveGeneric       #-}
-{-# LANGUAGE FlexibleContexts    #-}
-{-# LANGUAGE GADTs               #-}
-{-# LANGUAGE NamedFieldPuns      #-}
-{-# LANGUAGE OverloadedStrings   #-}
-{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE ConstraintKinds #-}
```
Comment (erikd) [inline cardano-api/src/Cardano/Api/Protocol.hs:1]: :100:

File: cardano-api/src/Cardano/Api/TextView.hs [hunk 2]
```diff
@@ -51,11 +50,12 @@ newtype TextViewDescription
 -- It also contains a \"title\" field which is free-form, and could be used
 -- to indicate the role or purpose to a reader.
 --
-data TextView = TextView
-  { tvType :: !TextViewType
-  , tvDescription :: !TextViewDescription
-  , tvRawCBOR :: !ByteString
-  } deriving (Eq, Show)
+data TextView
+  = TextView { tvType        :: !TextViewType
```
Comment (erikd) [inline cardano-api/src/Cardano/Api/TextView.hs:54]: Would be nice to fix:
* unneeded new line for the `= TextView`
* insane vertical alignment.

File: cardano-api/src/Cardano/Api/TxSubmit/ErrorRender.hs [hunk 3]
```diff
@@ -8,31 +8,31 @@ module Cardano.Api.TxSubmit.ErrorRender
...
-import           Formatting ((%), build, sformat, stext)
+import           Formatting (build, sformat, stext, (%))
 
 renderApplyMempoolPayloadErr :: ApplyMempoolPayloadErr -> Text
 renderApplyMempoolPayloadErr err =
     case err of
-      MempoolTxErr ve -> renderValidationError ve
-      MempoolDlgErr {} -> "Delegation error"
+      MempoolTxErr ve             -> renderValidationError ve
+      MempoolDlgErr {}            -> "Delegation error"
```
Comment (erikd) [inline cardano-api/src/Cardano/Api/TxSubmit/ErrorRender.hs:26]: Vertial alignment (other than that needed by indentation rules) is bad!

File: cardano-api/src/Cardano/Api/TxSubmit/Types.hs [hunk 4]
```diff
@@ -20,13 +20,13 @@ import qualified Data.Aeson as Aeson
...
 
 
-data NodeApiEnv = NodeApiEnv
-  { naeConfig :: Genesis.Config
-  , naeSocket :: SocketPath
-  }
+data NodeApiEnv
+  = NodeApiEnv { naeConfig :: Genesis.Config
+               , naeSocket :: SocketPath
+               }
```
Comment (erikd) [inline cardano-api/src/Cardano/Api/TxSubmit/Types.hs:29]: Would be good to get something as close to what we had as possible.

File: cardano-api/src/Cardano/Api/TxSubmit/ErrorRender.hs (hunk 3 above)
Comment (dcoutts) [reply cardano-api/src/Cardano/Api/TxSubmit/ErrorRender.hs:26]: :sob:

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 5]
```diff
@@ -1787,23 +1775,12 @@ toShelleyPoolParams StakePoolParameters {
...
-       -- | A URL to a homepage with additional information about the pool.
-       -- n.b. the spec does not specify a character limit for this field.
-     , stakePoolHomepage :: !Text
-     }
+data StakePoolMetadata
+  = StakePoolMetadata { stakePoolName        :: !Text
+                      , stakePoolDescription :: !Text
+                      , stakePoolTicker      :: !Text
+                      , stakePoolHomepage    :: !Text
+                      }
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api/Typed.hs:1783]: All the field comments got accidentally deleted here.

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 6]
```diff
@@ -1981,119 +1958,25 @@ instance HasTypeProxy UpdateProposal where
...
+                             , protocolUpdateMinUTxOValue        :: Maybe Lovelace
+                             , protocolUpdateStakeAddressDeposit :: Maybe Lovelace
+                             , protocolUpdateStakePoolDeposit    :: Maybe Lovelace
+                             , protocolUpdateMinPoolCost         :: Maybe Lovelace
+                             , protocolUpdatePoolRetireMaxEpoch  :: Maybe EpochNo
+                             , protocolUpdateStakePoolTargetNum  :: Maybe Natural
+                             , protocolUpdatePoolPledgeInfluence :: Maybe Rational
+                             , protocolUpdateMonetaryExpansion   :: Maybe Rational
+                             , protocolUpdateTreasuryCut         :: Maybe Rational
+                             }
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api/Typed.hs:1979]: All the field haddock comments have accidentally been deleted.

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 7]
```diff
@@ -2378,26 +2260,12 @@ withNodeProtocolClient (CardanoMode epochSlots securityParam) f =
...
-                    block
-                    (Query block)
-                    IO ())
-     }
+data LocalNodeClientProtocols block
+  = LocalNodeClientProtocols { localChainSyncClient :: Maybe (ChainSyncClient block (Tip block) IO ())
+                             , localTxSubmissionClient :: Maybe
+  (LocalTxSubmissionClient (GenTx block) (ApplyTxErr block) IO ())
+                             , localStateQueryClient :: Maybe (LocalStateQueryClient block (Query block) IO ())
+                             }
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api/Typed.hs:2268]: Honestly this is much worse than before. Just look at it.

File: cardano-api/src/Cardano/Api/TxSubmit/ErrorRender.hs (hunk 3 above)
Comment (newhoggy) [reply cardano-api/src/Cardano/Api/TxSubmit/ErrorRender.hs]: I've disabled this rule.

File: cardano-api/src/Cardano/Api/Typed.hs (hunk 7 above)
Comment (newhoggy) [reply cardano-api/src/Cardano/Api/Typed.hs]: I've disabled record formatting altogether.

File: cardano-api/src/Cardano/Api/TextView.hs (hunk 2 above)
Comment (newhoggy) [reply cardano-api/src/Cardano/Api/TextView.hs]: I've disable record formatting altogether.

File: cardano-api/src/Cardano/Api/TxSubmit/Types.hs (hunk 4 above)
Comment (newhoggy) [reply cardano-api/src/Cardano/Api/TxSubmit/Types.hs]: I've disabled record formatting altogether.

**Pull request merged**